### PR TITLE
[TT-15674] Extend pumps config with ca cert option

### PR DESCRIPTION
**Pull request by vladzabolotnyi**
## Description

[TT-15674](https://tyktech.atlassian.net/browse/TT-15674)

Adds SSL CA certificate file configuration support to Kafka, Elasticsearch, and Splunk pumps for consistency with existing MongoDB pump functionality.

**New Configuration:**
- `ssl_ca_file` meta property for specifying custom CA certificate file paths
- Environment variables:
  - `TYK_PMP_PUMPS_KAFKA_META_SSLCAFILE`
  - `TYK_PMP_PUMPS_ELASTICSEARCH_META_SSLCAFILE`
  - `TYK_PMP_PUMPS_SPLUNK_META_SSLCAFILE`

**Key Features:**
- Validates CA certificates using Go stdlib before applying TLS configuration
- Maintains backward compatibility - feature is optional
- Logs warnings for configuration conflicts (e.g., when `ssl_insecure_skip_verify` is enabled)
- Comprehensive error handling for file read errors and invalid PEM formats

## Motivation and Context

Currently, only MongoDB pumps support custom CA certificate configuration. This creates inconsistency across pumps and limits users' ability to connect to services using self-signed certificates or custom CAs. This change brings Kafka, Elasticsearch, and Splunk pumps to feature parity with MongoDB.

## How This Has Been Tested

- **Local Testing**: Validated against Docker containers running Elasticsearch and Splunk using their CA certificates
- **Unit Tests**: Added comprehensive tests covering CA loading logic, error scenarios, and edge cases
- **Integration**: Verified TLS connections work correctly with custom CA files

The implementation focuses on configuration validation and TLS setup rather than end-to-end pump functionality, making thorough unit testing sufficient for validation.

### Demo

_Success case with valid CA cert for Splunk server_

https://github.com/user-attachments/assets/b7317591-3248-49f0-b7b3-869ee17c1200

_Fail case when CA cert value was changed with invalid value_

https://github.com/user-attachments/assets/3ce8af90-b574-4af5-8f23-1d6dfde49228

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

## Checklist
- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side)
- [x] Make sure you are making a pull request against the **`master` branch** (left side)
- [x] My change requires a change to the documentation
  - [x] Documentation updates for new environment variables and configuration options
- [x] I have updated the documentation accordingly
- [x] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] Check your code additions will not fail linting checks:
  - [x] `go fmt -s`
  - [x] `go vet`

## Additional Notes

- **Breaking Changes**: None - all new configuration is optional
- **Documentation**: README.md and pump-specific documentation need updates for new environment variables
- **Security**: CA certificates are validated before use, and clear error messages guide users through configuration issues

[TT-15674]: https://tyktech.atlassian.net/browse/TT-15674?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### Ticket Details

<details>
<summary>
<a href="https://tyktech.atlassian.net/browse/TT-15674" title="TT-15674" target="_blank">TT-15674</a>
</summary>

|         |    |
|---------|----|
| Status  | In Code Review |
| Summary | [Pump] Configure CA file via environment variable |

Generated at: 2026-02-05 11:16:04

</details>

### CODE REVIEW

File: pumps/elasticsearch.go [hunk 1]
```diff
@@ -82,6 +81,8 @@ type ElasticsearchConf struct {
 	SSLCertFile string `json:"ssl_cert_file" mapstructure:"ssl_cert_file"`
 	// Can be used to set custom key file for authentication with Elastic Search.
 	SSLKeyFile string `json:"ssl_key_file" mapstructure:"ssl_key_file"`
+	// Can be used to set the file with CA certificates for verifying the Elastic Search server's certificate.
```
Comment (andyo-tyk) [inline pumps/elasticsearch.go]:
```suggestion
	// Path to the PEM file with trusted CA certificates that will be used to verify the Elasticsearch server's certificate.
```

File: README.md [hunk 2]
```diff
@@ -895,6 +913,7 @@ More advanced fields:
 - `meta_data`: Can be used to set custom metadata inside the kafka message
 - `ssl_cert_file`: Can be used to set custom certificate file for authentication with kafka.
 - `ssl_key_file`: Can be used to set custom key file for authentication with kafka.
```
Comment (andyo-tyk) [inline README.md]:
```suggestion
- `ssl_key_file`: Path to the PEM file with Tyk's private key for authentication with the Kafka server.
```

**Review by kofoworola — APPROVED** [review]
